### PR TITLE
Add metric for tmax timeout exceed count

### PR DESCRIPTION
**Pull request by Sonali-More-Xandr**
We have added Tmax.BidderResponseDurationMin check in auction workflow. As per this change, PBS won't send request to bidder server if remaining duration less than BidderResponseDurationMin. This PR adds a counter metric for such rejected request.

### CODE REVIEW

File: metrics/prometheus/prometheus.go [hunk 1]
```diff
@@ -197,6 +198,10 @@ func NewMetrics(cfg config.PrometheusMetrics, disabledMetrics config.DisabledMet
 		"connections_opened",
 		"Count of successful connections opened to Prebid Server.")
 
+	metrics.tmaxTimeout = newCounterWithoutLabels(cfg, reg,
+		"tmax_timeout",
+		"Count of requests rejected due to timeout exceed.")
```
Comment (onkarvhanumante) [inline metrics/prometheus/prometheus.go:203]: lets add tmax in description. How about `Count of requests rejected due to Tmax timeout exceed.`

File: metrics/go_metrics.go [hunk 2]
```diff
@@ -288,6 +289,7 @@ func getModuleNames(moduleStageNames map[string][]string) []string {
 func NewMetrics(registry metrics.Registry, exchanges []openrtb_ext.BidderName, disableAccountMetrics config.DisabledMetrics, syncerKeys []string, moduleStageNames map[string][]string) *Metrics {
 	newMetrics := NewBlankMetrics(registry, exchanges, disableAccountMetrics, moduleStageNames)
 	newMetrics.ConnectionCounter = metrics.GetOrRegisterCounter("active_connections", registry)
+	newMetrics.TMaxTimeoutCounter = metrics.GetOrRegisterCounter("tmax_timeout", registry)
```
Comment (onkarvhanumante) [inline metrics/go_metrics.go:292]: should update unit test and add test to check counter value

File: exchange/bidder.go [hunk 3]
```diff
@@ -539,6 +539,7 @@ func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.Re
 
 	if tmaxAdjustments != nil && tmaxAdjustments.Enabled && tmaxAdjustments.BidderResponseDurationMin > 0 {
 		if hasShorterDurationThanTmax(&bidderTmaxCtx{ctx}, *tmaxAdjustments) {
+			bidder.me.RecordTMaxTimeout()
```
Comment (onkarvhanumante) [inline exchange/bidder.go:542]: lets update `doRequestImpl` unit tests for this

File: metrics/prometheus/prometheus.go (hunk 1 above)
Comment (Sonali-More-Xandr) [reply metrics/prometheus/prometheus.go]: updated description in [62c4b50](https://github.com/prebid/prebid-server/pull/2892/commits/62c4b507afbb8af6aabdb3e20e6cb3ec245f4761)

File: metrics/go_metrics.go (hunk 2 above)
Comment (Sonali-More-Xandr) [reply metrics/go_metrics.go:292]: added a test in [4e21c81](https://github.com/prebid/prebid-server/pull/2892/commits/4e21c81beed83e49adff76494a1b74c674a589b9)

File: exchange/bidder.go (hunk 3 above)
Comment (Sonali-More-Xandr) [reply exchange/bidder.go:542]: Updated in [e44b312](https://github.com/prebid/prebid-server/pull/2892/commits/e44b3125370ed31439834b14175d8b7d046c25d3)

**Comment by guscarreon** [comment]
Hi @Sonali-More-Xandr . This pull request has been unblocked but there are conflicts that must be resolved.

**Review by hhhjort — APPROVED** [review]
LGTM

**Pull request merged**